src/Common.sol

```diff
@@ -0,0 +1,537 @@
+                params.token0,
+                params.token1,
+                params.tickLower,
+                params.tickUpper,
+                params.amount0Desired,
+                params.amount1Desired,
+                params.amount0Min,
+                params.amount1Min,
+                address(this), // is sent to real recipient aftwards
+                params.deadline
+            );
+
+        // mint is done to address(this) because it is not a safemint and safeTransferFrom needs to be done manually afterwards
+        return nfpm.mint(mintParams);
+    }
+
+    // swap and increase logic
+    function _swapAndIncrease(SwapAndIncreaseLiquidityParams memory params, IERC20 token0, IERC20 token1, bool unwrap) internal returns (uint128 liquidity, uint256 added0, uint256 added1) {
+        (uint256 total0, uint256 total1) = _swapAndPrepareAmounts(
+            SwapAndMintParams(params.protocol, params.nfpm, token0, token1, 0, 0, 0, params.amount0, params.amount1, params.recipient, params.deadline, params.swapSourceToken, params.amountIn0, params.amountOut0Min, params.swapData0, params.amountIn1, params.amountOut1Min, params.swapData1, params.amountAddMin0, params.amountAddMin1, ""), unwrap);
+
+        INonfungiblePositionManager.IncreaseLiquidityParams memory increaseLiquidityParams = 
+            univ3.INonfungiblePositionManager.IncreaseLiquidityParams(
+                params.tokenId, 
+                total0, 
+                total1, 
+                params.amountAddMin0,
+                params.amountAddMin1, 
+                params.deadline
+            );
+
+        (liquidity, added0, added1) = params.nfpm.increaseLiquidity(increaseLiquidityParams);
+
+        emit SwapAndIncreaseLiquidity(params.tokenId, liquidity, added0, added1);
+        IWETH9 weth = _getWeth9(params.nfpm, params.protocol);
+        _returnLeftoverTokens(ReturnLeftoverTokensParams(weth, params.recipient, token0, token1, total0, total1, added0, added1, unwrap));
+    }
+
+    // swaps available tokens and prepares max amounts to be added to nfpm
+    function _swapAndPrepareAmounts(SwapAndMintParams memory params, bool unwrap) internal returns (uint256 total0, uint256 total1) {
+        if (params.swapSourceToken == params.token0) { 
+            if (params.amount0 < params.amountIn1) {
+                revert AmountError();
+            }
+            (uint256 amountInDelta, uint256 amountOutDelta) = _swap(params.token0, params.token1, params.amountIn1, params.amountOut1Min, params.swapData1);
+            total0 = params.amount0 - amountInDelta;
+            total1 = params.amount1 + amountOutDelta;
+        } else if (params.swapSourceToken == params.token1) { 
+            if (params.amount1 < params.amountIn0) {
+                revert AmountError();
+            }
+            (uint256 amountInDelta, uint256 amountOutDelta) = _swap(params.token1, params.token0, params.amountIn0, params.amountOut0Min, params.swapData0);
```

Comment:
this should be 
`
_swap(params.token1, params.token0, params.amountIn1, params.amountOut1Min, params.swapData1);
`
following the logic in V3Utils 
`
_swap(IERC20(token0), IERC20(instructions.targetToken), amount0, instructions.amountOut0Min, instructions.swapData0);
`
same for swap from `token0`

Reply:
It's not a bug. It's a feature